### PR TITLE
Feature/tr 6795/external qti previewer 2

### PR DESCRIPTION
**Pull request by olga-kulish**
https://oat-sa.atlassian.net/browse/TR-6793

For https://github.com/oat-sa/extension-tao-testqti-previewer/pull/243

### Changes
- return `remainingAttempts` & `validateResponses`

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved preview configuration validation and provider selection with better fallbacks and conditional inclusion of proxy settings.
  * Updated preview mapping to honor per-item session controls, producing correct remaining-attempt counts and per-item response-validation flags.

* **Tests**
  * Expanded unit tests to cover session-control variations and validate mapping of new per-item fields.

### CODE REVIEW

File: models/test/mapper/TestPreviewMapper.php [hunk 1]
```diff
@@ -92,9 +92,13 @@ public function map(AssessmentTest $test, Route $route, TestPreviewConfig $confi
                 $itemUri = $itemRef->getHref();
 
                 $allowSkipping = true;
+                $validateResponses = false;
+                $remainingAttempts = -1;
                 $sessionControl = $itemRef->getItemSessionControl();
                 if ($sessionControl !== null) {
                     $allowSkipping = $sessionControl->doesAllowSkipping();
+                    $validateResponses = $sessionControl->mustValidateResponses();
+                    $remainingAttempts = $sessionControl->getMaxAttempts() - 1;
```
Comment (kilatib) [inline models/test/mapper/TestPreviewMapper.php:101]: Not sure about that, can you please provide more details here

File: controller/Previewer.php [hunk 2]
```diff
@@ -335,9 +335,17 @@ public function configuration(): void
         }
 
         try {
+            $runnerProviders = $response['value']['providers'];
+            $previewProviders = $response['value']['previewProviders'] ?? [];
+            $providers = [];
+            foreach (['runner', 'itemRunner', 'plugins'] as $key) {
+                $providers[$key] = $previewProviders[$key] ?? $runnerProviders[$key];
```
Comment (oatymart) [inline controller/Previewer.php:342]: I had thought we'd manage this by making this endpoint compatible with `previewProviders` only, and compose the contents of `previewProviders` at the configuration level (in NGS and k8s it's quite easy to design the structure reusing certain parts from `providers`).

I suppose your version will give the same end result, and saves us from having that composition done in multiple places.

Comment (olga-kulish) [reply controller/Previewer.php]: When we would need to remove security plugins in previewer, do you think it should be done in k8s config?

I planned to do it in deliver-fe, because deliver-fe understands what it can support in previewer and what it can not support.
And if that is done in deliver-fe, `providers` and `previewProviders` will be identical, there's no need to separate them unless some unknown edge case will appear.

If you think plugins should be removed in k8s, ok for me too. Then we can make `previewProviders` section with all its subsections) mandatory.

File: models/test/mapper/TestPreviewMapper.php (hunk 1 above)
Comment (olga-kulish) [reply models/test/mapper/TestPreviewMapper.php:101]: This data needs to be passed to test-runner frontend. `getMaxAttempts() - 1` not `getMaxAttempts()` is because this is what happens in real delivery - first attempt is already started, so remaining attempts are `-1`.

**Review by kilatib — APPROVED** [review]
Looks good, as for me GJ!

File: controller/Previewer.php (hunk 2 above)
Comment (oatymart) [reply controller/Previewer.php]: If possible, I think we could do without this layer of logic.

We will have [definitions in tenant-data.libsonnet] -> [logic in Previewer.php] -> [logic in previewer-app after fetching configuration].
That's 3 layers, but if this backend layer would just be a pass-through for `previewerProviders`, it would be effectively 2 layers. So less code, and less chance of needing to re-release code.

We can start with the simplest possible tenant-data.libsonnet: `previewProviders: self.providers + { itemRunner: { ... }, proxy: { ... } }`
and as soon as we need to customise it further, we will.

Comment (olga-kulish) [reply controller/Previewer.php]: ~~Ok, as I understand, `previewProviders` is just a pass-through, and removing/updating unsupported plugins/options is still done on deliver-fe. That's indeed much cleaner.~~ 

Fixed.

**Review by oatymart — APPROVED** [review]
